src/xo-app/backup/restore/index.js

```diff
@@ -98,16 +103,90 @@ const doImport = ({ backup, sr, start }) => {
   }
 }
 
+class _Collapsible extends Component {
+  render () {
+    const {
+      collapsible,
+      children,
+      ...props
+    } = this.props
+
+    return collapsible
+      ? <Collapse {...props}> {children} </Collapse>
+      : <div>
+        <span> {props.buttonText} </span>
+        <br />
+        {children}
+      </div>
+  }
+}
+
 class _ModalBody extends Component {
+  constructor () {
+    super()
+
+    this.state = {
+      mapVdisSrs: {}
+    }
+  }
+
   get value () {
     return this.state
   }
 
+  _getSrPredicate = createSelector(
+    () => this.state.sr,
+    () => this.state.mapVdisSrs,
+    (defaultSr, mapVdisSrs) => sr => {
+      let predicate = isSrWritable(sr) && defaultSr.$pool === sr.$pool
+
+      if (!defaultSr.shared) {
+        predicate = predicate && (defaultSr.$container === sr.$container || sr.shared)
+      } else {
+        forEach(mapVdisSrs, (selectedSr, vdi) => {
+          if (selectedSr != null && !selectedSr.shared) {
+            predicate = predicate && (selectedSr.$container === sr.$container || sr.shared)
+          }
+        })
+      }
+
+      return predicate
+    }
+  )
+
+  _onChangeDefaultSr = event => {
+    const oldSr = this.state.sr
+    const newSr = getEventValue(event)
+
+    if (oldSr != null && newSr != null) {
+      if (oldSr.$pool !== newSr.$pool) {
+        this.setState({
+          mapVdisSrs: {}
```

Comment:
Don't you also want to reinitialize `mapVdisSrs` when `oldSr == null` or `newSr == null`?